### PR TITLE
dts: bindings: Update DMA bindings with DMA config macros

### PR DESCRIPTION
**Pull request by Desvauxm-st**
Update DMA bindings with DMA config macros
only for st,stm32-dma-v2.yaml and st,stm32-dma-v2bis.yaml

### CODE REVIEW

File: dts/bindings/dma/st,stm32-dma-v2.yaml [hunk 1]
```diff
@@ -17,35 +17,62 @@ description: |
...
-               0x0: Byte (8 bits)
-               0x1: Half-word (16 bits)
-               0x2: Word (32 bits)
-               0x3: reserved
-        -bit 15: Reserved
-        -bit 16-17 : Priority level
-               0x0: low
-               0x1: medium
-               0x2: high
-               0x3: very high
```
Comment (erwango) [inline dts/bindings/dma/st,stm32-dma-v2.yaml:48]: Please keep this, it is still valuable, clear and readable information.

File: dts/bindings/dma/st,stm32-dma-v2.yaml [hunk 2]
```diff
@@ -17,35 +17,62 @@ description: |
...
+    3. channel-config: A name custom DMA flags for channel configuration
+    which is device dependent see stm32_dma.h:
+       /** DMA  cyclic mode config on bit 5*/
+       #define STM32_DMA_CH_CFG_MODE(val)		((val & 0x1) << 5)
+       #define STM32_DMA_MODE_NORMAL		STM32_DMA_CH_CFG_MODE(0)
+       #define STM32_DMA_MODE_CYCLIC		STM32_DMA_CH_CFG_MODE(1)
```
Comment (erwango) [inline dts/bindings/dma/st,stm32-dma-v2.yaml:25]: I don't see great interest in copy/pasting headers. People can open them directly.

File: dts/bindings/dma/st,stm32-dma-v2.yaml (hunk 1 above)
Comment (erwango) [reply dts/bindings/dma/st,stm32-dma-v2.yaml:48]: What about:
        -bit 6-7 : Direction  (see dma.h)
               0x0: STM32_DMA_MEMORY_TO_MEMORY: MEM to MEM
               0x1: STM32_DMA_MEMORY_TO_PERIPH: MEM to PERIPH                              
               0x2: STM32_DMA_PERIPH_TO_MEMORY: PERIPH to MEM
               0x3: reserved for PERIPH to PERIPH

File: dts/bindings/dma/st,stm32-dma-v2.yaml [hunk 3]
```diff
@@ -17,35 +17,62 @@ description: |
     this value is 0 for Memory-to-memory transfers
     or a value between <1> .. <dma-generators> (not supported yet)
     or a value beweeen <dma-generators>+1  ..  <dma-generators>+<dma-requests>
-    3. channel-config: A 32bit mask specifying the DMA channel configuration
```
Comment (erwango) [inline dts/bindings/dma/st,stm32-dma-v2.yaml:20]: Please keep this, it is still valuable, clear and readable information.

**Review by erwango — CHANGES_REQUESTED** [review]

File: dts/bindings/dma/st,stm32-dma-v2.yaml (hunk 3 above)
Comment (Desvauxm-st) [reply dts/bindings/dma/st,stm32-dma-v2.yaml:20]: OK

File: dts/bindings/dma/st,stm32-dma-v2.yaml (hunk 2 above)
Comment (Desvauxm-st) [reply dts/bindings/dma/st,stm32-dma-v2.yaml]: ok

**Review by FRASTM — APPROVED** [review]
Rmk : Requires the dts file to include the ./include/zephyr/dt-bindings/dma/stm32_dma.h

**Review by erwango — APPROVED** [review]

**Pull request merged**